packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -478,18 +491,12 @@ async function listAndCountProductWithIsolatedProductModule(
     metadata: { count },
   } = await remoteQuery(query, variables)
 
-  // TODO: Waiting for pr #4990 before being able to remove that logic and replace it by a new relation in the query above
-  const shippingProfileService = req.scope.resolve(
-    "shippingProfileService"
-  ) as ShippingProfileService
-
-  const profilesByProductId = await shippingProfileService.retrieveForProducts(
-    products.map((p) => p.id)
-  )
-
   products.forEach((product) => {
-    product.profile_id = profilesByProductId[product.id]?.[0]?.id
-    product.profiles = profilesByProductId[product.id]
+    product.profile_id = product.shipping_profile?.profile?.id
```

Comment:
Yes we discussed with carlos about it, i ll beta test it

Reply:
I ve fixed one part of the remote joiner, but I need to discuss a point with @carlos-r-l-rodrigues before being able to use it in this pr

Reply:
Sure, what's stopping you from including it now? I am just curious

Reply:
waiting that pr to be merge @olivermrbl https://github.com/medusajs/medusa/pull/5023

Reply:
Hmm, I don't think this is what we want.

What you ask for in the "gql" string is the exact response, so, in this case, I would assume the `profile_id` to be living at `product.profile.id`.

Is this possible without jumping through hoops?

Reply:
Otherwise, developers will need to dig through the field aliases to find the property path of what they queried using the Remote Query. Does this make sense?

I realise we will not be exposing this tool publicly in the near future, but I think it is worth thinking about it. If it's not possible without jumping through too many hoops, we can punt this discussion for later.